### PR TITLE
Shoot care controller garbage-collects orphaned `Lease` objects related to no longer existing `Node`s

### LINKED ISSUE #8749: Node leases never get cleaned up
**How to categorize this issue?**
/area control-plane

/kind cleanup
/kind bug

**What happened**:
kubelet creates a node lease for every node in the `kube-node-lease` namespace. We have noticed that across shoot clusters, node leases are never cleaned up and pile up.
Example:
Take aws-ha (dev landscape) seed as an example
Total number of nodes: 9
Total number of node leases: 388

For leases which do not have any node, their `OwnerReference` is absent but their `holderIdentity` still has the node name.
There is a related upstream issue https://github.com/kubernetes/kubernetes/issues/119660 

**What you expected to happen**:
Node leases should be eventually cleaned up after their parent nodes are no longer available. A potential enhancement to garbage collection is required.

**How to reproduce it (as minimally and precisely as possible)**:
Just look at any shoot cluster.

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area scalability
/kind enhancement

**What this PR does / why we need it**:
Shoot care controller garbage-collects orphaned `Lease` objects related to no longer existing `Node`s, see https://github.com/kubernetes/kubernetes/issues/119660.

**Which issue(s) this PR fixes**:
Fixes #8749

**Special notes for your reviewer**:
/cc @timuthy 

**Release note**:

```feature operator
`gardenlet'`s `Shoot` care controller now garbage-collects orphaned `Lease` objects related to no longer existing `Node`s - see [this upstream issue](https://github.com/kubernetes/kubernetes/issues/119660) for more details.
```

### CODE REVIEW

File: pkg/gardenlet/controller/shoot/care/garbage_collection.go [hunk 1]
```diff
@@ -84,34 +88,64 @@ func (g *GarbageCollection) Collect(ctx context.Context) {
...
-	podList := &corev1.PodList{}
-	if err := shootClient.List(ctx, podList, client.InNamespace(namespace)); err != nil {
+	return g.deleteStalePods(ctx, shootClient, namespace)
+}
+
+// See https://github.com/gardener/gardener/issues/8749 and https://github.com/kubernetes/kubernetes/issues/109777.
+// kubelet sometimes created Lease objects without owner reference. When the respective node gets deleted eventually,
+// the Lease object remains in the system and no Kubernetes controller will ever clean it up. Hence, this function takes
+// over this task.
+func (g *GarbageCollection) deleteOrphanedNodeLeases(ctx context.Context, c client.Client) error {
```
Comment (acumino) [inline pkg/gardenlet/controller/shoot/care/garbage_collection.go:114]: TODO to drop with end of  1.28 support?

**Review by acumino — COMMENTED** [review]
Thanks!
/lgtm

**Review by acumino — COMMENTED** [review]
Thanks!
/lgtm

File: pkg/gardenlet/controller/shoot/care/garbage_collection.go [hunk 2]
```diff
@@ -84,34 +88,65 @@ func (g *GarbageCollection) Collect(ctx context.Context) {
...
+		}
+		lease := l.DeepCopy()
+
+		if err := c.Get(ctx, client.ObjectKey{Name: lease.Name}, &metav1.PartialObjectMetadata{TypeMeta: metav1.TypeMeta{APIVersion: corev1.SchemeGroupVersion.String(), Kind: "Node"}}); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return fmt.Errorf("failed getting node %s when checking for potential orphaned Lease %s: %w", lease.Name, client.ObjectKeyFromObject(lease), err)
+			}
+
+			g.log.Info("Detected orphaned Lease object, cleaning it up", "nodeName", lease.Name, "lease", client.ObjectKeyFromObject(lease))
+			orphanedLeases = append(orphanedLeases, lease.DeepCopy())
```
Comment (timuthy) [inline pkg/gardenlet/controller/shoot/care/garbage_collection.go]: Why another copy?

**Review by timuthy — COMMENTED** [review]
Nice 👏

**Review by timuthy — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**